### PR TITLE
#1647-ExecutorSpec #1

### PR DESCRIPTION
**Pull request by XLPI**
Please review ExecutorSpec migration to zio

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala [hunk 1]
```diff
@@ -35,9 +37,9 @@ object TestExecutor {
   }
 }
 
-class ExecutorSpec extends Specification {
+object SpecDescription {
```
Comment (regiskuckaertz) [inline core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala]: You can drop this entire object, the `is` function is a specs2 artefact.

Comment (XLPI) [reply core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala]: should I maybe somewhere put description what actually tests does?

File: core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala [hunk 2]
```diff
@@ -35,44 +37,51 @@ object TestExecutor {
...
-  def yield1 = TestExecutor.y.submitOrThrow(() => ()) must not(throwA[RejectedExecutionException])
-  def yield2 = TestExecutor.y.asEC.execute(() => ()) must not(throwA[RejectedExecutionException])
-  def yield3 = Executor.fromExecutionContext(1)(TestExecutor.ec).submit(() => ()) must beTrue
-
-  def unyield1 = TestExecutor.u.submitOrThrow(() => ()) must not(throwA[RejectedExecutionException])
-  def unyield2 = TestExecutor.u.asEC.execute(() => ()) must not(throwA[RejectedExecutionException])
-}
+object ExecutorSpec
+    extends ZIOBaseSpec(
+      suite("Create the default unyielding executor and check that:")(
```
Comment (ghostdogpr) [inline core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala]: Can you make the root suite just called`ExecutorSpec` and contains the other suites?

Comment (XLPI) [reply core-tests/shared/src/test/scala/zio/internal/ExecutorSpec.scala]: ok, I'll do it

**Comment by XLPI** [comment]
@regiskuckaertz please take a look again (also fix by @ghostdogpr comments)

**Comment by ghostdogpr** [comment]
Currently the suites are like this:
```
[info] + ExecutorSpec
[info]   + Create the default unyielding executor and check that:
[info]     + When converted to an EC, it reports Throwables to stdout
[info]     + Create an executor that cannot have tasks submitted to and check that:
[info]       + It throws an exception upon submission
[info]       + When converted to an ExecutionContext, it throws an exception
[info]       + When created from an EC, throw when fed an effect
[info]     + Create a yielding executor and check that:
[info]       + Runnables can be submitted 
[info]       + When converted to an ExecutionContext, it accepts Runnables
[info]       + When created from an EC, must not throw when fed an effect 
[info]     + Create an unyielding executor and check that:
[info]       + Runnables can be submitted
[info]       + When converted to an ExecutionContext, it accepts Runnables
```
But it should be:
```
[info] + ExecutorSpec
[info]   + Create the default unyielding executor and check that:
[info]     + When converted to an EC, it reports Throwables to stdout
[info]   + Create an executor that cannot have tasks submitted to and check that:
[info]     + It throws an exception upon submission
[info]     + When converted to an ExecutionContext, it throws an exception
[info]     + When created from an EC, throw when fed an effect
[info]   + Create a yielding executor and check that:
[info]     + Runnables can be submitted 
[info]     + When converted to an ExecutionContext, it accepts Runnables
[info]     + When created from an EC, must not throw when fed an effect 
[info]   + Create an unyielding executor and check that:
[info]     + Runnables can be submitted
[info]     + When converted to an ExecutionContext, it accepts Runnables
```

**Comment by XLPI** [comment]
Oh, @ghostdogpr , I'll fix it

**Comment by XLPI** [comment]
@ghostdogpr  done!

**Comment by XLPI** [comment]
UPD: FIxed for backward compatibility for Scala 2.11.12

**Comment by XLPI** [comment]
@regiskuckaertz  can we merge?

**Comment by regiskuckaertz** [comment]
Yes! Congratulations on your first PR 🎉 on to the next one 💪

**Comment by XLPI** [comment]
WOW! @regiskuckaertz thanks! It was fascinatingly, interesting and amazing!

**Comment by ghostdogpr** [comment]
@XLPI 🎉

**Pull request merged**